From 6bf298d3620c286e478395fd8cb26602ef90922d Mon Sep 17 00:00:00 2001
From: Tony Cheng <tony.cheng@amd.com>
Date: Sat, 17 Sep 2016 02:14:58 -0400
Subject: [PATCH 1101/1722] drm/amd/dal: refactor aux 2

- add pointer to static table in hw object struct
- assign register offset table in ctor
- add reg_helper macro, with CTX and REG defined
- use macro to update 1 register

Change-Id: Idde34e0f55e94982cb986488f27a07549ae23f71
Signed-off-by: Tony Cheng <tony.cheng@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 .../amd/dal/dc/i2caux/dce110/aux_engine_dce110.c   | 22 ++++++++++------------
 .../amd/dal/dc/i2caux/dce110/aux_engine_dce110.h   |  2 ++
 .../drm/amd/dal/dc/i2caux/dce110/i2caux_dce110.c   |  3 ++-
 3 files changed, 14 insertions(+), 13 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
index f3f3f13..bbf33bc 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
@@ -44,6 +44,12 @@
 #include "dce/dce_11_0_d.h"
 #include "dce/dce_11_0_sh_mask.h"
 
+#define CTX \
+	aux110->base.base.ctx
+#define REG(reg_name)\
+	(aux110->regs->reg_name)
+#include "reg_helper.h"
+
 /*
  * This unit
  */
@@ -67,19 +73,9 @@
 static void release_engine(
 	struct engine *engine)
 {
-	struct aux_engine_dce110 *aux_engine = FROM_ENGINE(engine);
-
-	const uint32_t addr = aux_engine->addr.aux_arb_control;
+	struct aux_engine_dce110 *aux110 = FROM_ENGINE(engine);
 
-	uint32_t value = dm_read_reg(engine->ctx, addr);
-
-	set_reg_field_value(
-		value,
-		1,
-		AUX_ARB_CONTROL,
-		AUX_SW_DONE_USING_AUX_REG);
-
-	dm_write_reg(engine->ctx, addr, value);
+	REG_UPDATE(AUX_ARB_CONTROL, AUX_SW_DONE_USING_AUX_REG, 1);
 }
 
 static void destruct(
@@ -729,6 +725,8 @@ static bool construct(
 
 	engine->timeout_period = aux_init_data->timeout_period;
 
+	engine->regs = aux_init_data->regs;
+
 	return true;
 }
 
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.h b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.h
index 1a8e672..be59f59 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.h
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.h
@@ -47,6 +47,7 @@ struct dce110_aux_registers {
 
 struct aux_engine_dce110 {
 	struct aux_engine base;
+	const struct dce110_aux_registers *regs;
 	struct {
 		uint32_t aux_control;
 		uint32_t aux_arb_control;
@@ -62,6 +63,7 @@ struct aux_engine_dce110_init_data {
 	uint32_t engine_id;
 	uint32_t timeout_period;
 	struct dc_context *ctx;
+	const struct dce110_aux_registers *regs;
 };
 
 struct aux_engine *dal_aux_engine_dce110_create(
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2caux_dce110.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2caux_dce110.c
index fd965bf..39e1c7c 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2caux_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2caux_dce110.c
@@ -163,7 +163,7 @@ static const struct i2caux_funcs i2caux_funcs = {
 	AUX_COMMON_REG_LIST(id) \
 }
 
-static const struct dce110_aux_registers link_enc_aux_regs[] = {
+static const struct dce110_aux_registers aux_regs[] = {
 		aux_regs(0),
 		aux_regs(1),
 		aux_regs(2),
@@ -242,6 +242,7 @@ bool dal_i2caux_dce110_construct(
 		aux_init_data.engine_id = i;
 		aux_init_data.timeout_period = base->aux_timeout_period;
 		aux_init_data.ctx = ctx;
+		aux_init_data.regs = &aux_regs[i];
 
 		base->aux_engines[line_id] =
 			dal_aux_engine_dce110_create(&aux_init_data);
-- 
2.7.4

